packages/expo-cli/src/commands/build/AndroidBuilder.ts

```diff
@@ -213,12 +213,30 @@ See https://docs.expo.io/versions/latest/distribution/building-standalone-apps/#
   }
 
   checkEnv(): boolean {
-    return (
+    const allEnvSet =
       !!this.options.keystorePath &&
       !!this.options.keystoreAlias &&
       !!process.env.EXPO_ANDROID_KEYSTORE_PASSWORD &&
-      !!process.env.EXPO_ANDROID_KEY_PASSWORD
-    );
+      !!process.env.EXPO_ANDROID_KEY_PASSWORD;
+
+    if (allEnvSet) {
+      return true;
+    }
+
+    // Check if user was trying to upload keystore incorretly and supply an helpful error message if so.
+    if (!!this.options.keystorePath || !!this.options.keystoreAlias) {
```

Comment:
```suggestion
    if (this.options.keystorePath || this.options.keystoreAlias) {
```
Manually coercing these to boolean is not necessary, since it will happen automatically in boolean context. I think removing the `!!` makes the `if` statement slightly easier to scan quickly.